token/program-2022/src/instruction.rs

```diff
@@ -54,6 +56,8 @@ pub enum TokenInstruction {
     /// Otherwise another party can acquire ownership of the uninitialized
     /// account.
     ///
+    /// Any mixins must be initialized before this instruction.
```

Comment:
I'm a little squeamish about this initialization flow opening up new attack vectors, for both Mint and Accounts, but maybe it's ok.  The AccountType will still be uninitialized as mixins are getting initialized, so:
a. Mixin initialization must always fail if the account length is 355 (a possible multisig account), or less than 165 (legacy mint account)
b. My comment about "A single mixin type namespace for mints/accounts seems ok to me" probably needs to be stronger. There must be a single mixin type namespace: if I initialize an Account mixin for a Mint, then the Account mixin must be ignored.

Reply:
I'm with you on the attacks, but my thinking is that as long as the mint isn't properly initialized, it shouldn't be able to do damage.  For example, if I only initialize a bunch of mixins on a mint, but never do the proper `InitializeMint`, I can't create any token accounts on it.  We'll have to diligently check that the mint is initialized everywhere, but that's a given to me.

Otherwise, you're right, let's put it all in a single mixin namespace.  For now, all of the account mixins are initialized in `InitializeAccount`, so we should be ok from that front, since it's impossible to explicitly initialize an account mixin on its own, but we don't know if that'll hold forever.